### ISSUE TITLE
ASG max_queue_size resetting on cfncluster start

### ISSUE THREAD

**Issue by andrew80k** [awslabs/cfncluster#325]
We are running v1.3.1 and a solution to this issue `could` be to upgrade to 1.4.2, but I didn't see anything in the changelog that would cause me to assume this issue is resolved.

Nothing special about the environment, but here are the repeatable steps we used to recreate the issue.

1. create the cluster -- initial size of 2 nodes
2. resize the cluster -- doesn't matter to what but we used a resize to 10 
    (cfncluster update -nr -t tier03 MyGrid-Support)
[cfncluster-resize-issue1.txt](https://github.com/awslabs/cfncluster/files/1767708/cfncluster-resize-issue1.txt)

3. shutdown the grid 
    (cfncluster stop  MyGrid-Support)
[cfncluster-resize-issue2.txt](https://github.com/awslabs/cfncluster/files/1767710/cfncluster-resize-issue2.txt)

4. start the grid
    (cfncluster start  MyGrid-Support) 
[cfncluster-resize-issue3.txt](https://github.com/awslabs/cfncluster/files/1767711/cfncluster-resize-issue3.txt)

I have attached my cloud trail logs, but I can't seem to track down exactly where the part comes in where it sends that resize command.

relevant config pieces:
```
[cluster tier03]
vpc_settings = public-private
key_name = MyGrid_Grid_Master
template_url = https://s3.amazonaws.com/mygrid-cfncluster/cfncluster131/cfncluster-131fix-cfn.json
master_instance_type = t2.medium
compute_instance_type = c4.8xlarge
initial_queue_size = 0
max_queue_size = 10
scheduler = sge
cluster_type = ondemand
custom_ami = ami-fc89c8ea
#custom_ami = ami-4eae2e58
base_os = centos7
shared_dir = /shared
master_root_volume_size = 16
compute_root_volume_size = 16
scaling_settings = tier03
post_install = https://s3.amazonaws.com/mygrid-cfncluster/20170330-post-install/post_install.sh

[scaling tier03]
scaling_threshold = 1
scaling_adjustment = 5
scaling_threshold2 = 181
scaling_adjustment2 = 5
```

Any help, advice, suggestions would be appreciated.

**Comment by rajachan** [comment]
Andrew, thanks for creating the issue.  We [use the ASG to spin down the compute instances](https://github.com/awslabs/cfncluster/blob/develop/cli/cfncluster/cfncluster.py#L210) when a cluster is stopped (as opposed to terminating each instance manually). When a cluster is started again, the values from your config file are used to set the max_queue_size, among other ASG settings. Unless the config file was changed back to the default value between the stop and start, you should be seeing 10 after the restart. If the config was not changed after stopping the cluster, then we might have a bug. Let me know which one it is.

**Comment by andrew80k** [comment]
yep,  I could see that happening in the attached cluster-resize-issue2.txt file.  It set it to max size of 0, but then when we spun the grid back up, instead of using the tier03 config as it was set in cnfcluster update, it spun it up with max size of 2 instead of 10, see cluster-resize-issue3.txt.  The configuration did not change between the time it was stopped and the time it was restarted.

**Comment by rajachan** [comment]
Andrew, for the default cluster_template that is in the [global] section of your config, what is the max_queue_size set to? I'm guessing that's set to 2. We seem to have a bug in the cluster start logic that [selects the config section](https://github.com/awslabs/cfncluster/blob/develop/cli/cfncluster/cfnconfig.py#L101) to read the ASG values from.  I have some ideas on how to fix this. I'll leave this ticket open to track the bug. Until then, one workaround for this issue is to change the cluster_template in the global section to the template you want to use, in your case tier03.

**Comment by andrew80k** [comment]
@rajachan,

Could be as you say, though my global default section doesn't define a max size, it only points to a default template, which I have included as well. 

Here is my default global section:

```
[global]
update_check = true
sanity_check = true
cluster_template = default
```

cluster_template default:

```
[cluster default]
vpc_settings = public-private
key_name = MyGrid_Grid_Master
template_url = https://s3.amazonaws.com/mygrid-cfncluster/cfncluster131/cfncluster-131fix-cfn.json
master_instance_type = t2.medium
compute_instance_type = c4.large
initial_queue_size = 0
max_queue_size = 2
scheduler = sge
cluster_type = ondemand
custom_ami = ami-fc89c8ea
base_os = centos7
shared_dir = /shared
master_root_volume_size = 16
compute_root_volume_size = 16
post_install = https://s3.amazonaws.com/mygrid-cfncluster/20170330-post-install/post_install.sh
```

**Comment by rajachan** [comment]
Andrew - My guess was right. The global config section doesn't have to define a max_queue_size, but the "default" template that it points to does, and that is 2.  I am currently testing a fix, it should be available as part of the next release.

**Comment by andrew80k** [comment]
Thanks, @rajachan.  We look forward to the fix.

**Comment by andrew80k** [comment]
@rajachan Any idea when this fix might go in?

**Comment by rajachan** [comment]
I had just reworked the fix a bit based on some testing. The fix will be merged to the `develop` branch as soon as it gets reviewed and will be part of the next release.

**Comment by andrew80k** [comment]
Thanks for the update @rajachan.  Is there a schedule for the next release?
